src/gt4py/core.py

```diff
@@ -0,0 +1,18 @@
+# GT4Py - GridTools Framework
+#
+# Copyright (c) 2014-2024, ETH Zurich
+# All rights reserved.
+#
+# Please, refer to the LICENSE file in the root directory.
+# SPDX-License-Identifier: BSD-3-Clause
+
+from gt4py._core.definitions import CUPY_DEVICE_TYPE, Device, DeviceType, Scalar, is_scalar_type
+
+
+__all__ = [
+    "CUPY_DEVICE_TYPE",
+    "Device",
+    "DeviceType",
+    "Scalar",
```

Comment:
`Scalar` would be purely for typing, should we create a `typing` module? It would be a bit confusing with other `typing` modules as they I believe are pure typing, while here I would export the real thing. Other example is `decorator.Program` which should not be used by the user at runtime, but is needed for typing.